cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -72,6 +72,10 @@ struct ivf_pq_params {
 };
 
 using nn_descent_params = cuvs::neighbors::nn_descent::index_params;
+
+// **** Experimental ****
+using iterative_search_params = cuvs::neighbors::search_params;
```

Comment:
As I understand, `cuvs::neighbors::search_params` is being re-used here as an empty tag for the `std::variant` index parameters below.
While this is a perfectly valid approach, I wonder  whether it would look confusing to a user and whether it would cause us any troubles in future (if at some point we decide to add any members to `cuvs::neighbors::search_params`).
An alternative would be to just replace it with `struct iterative_search_params {}`.
What do you think, @cjnolet ?